### ISSUE TITLE
Unable to add any permissions to a function

### ISSUE THREAD

**Issue by tomchiverton** [architect/architect#1212]
**Describe the issue**
per https://arc.codes/docs/en/reference/config.arc/policies it should be trivial to apply existing SAM or custom policies to a function by configuring src/http/(name)/config.arc

None of these work

```
@aws
runtime nodejs12.x
policies S3CrudPolicy
```

```
@aws
runtime nodejs12.x
policies
   S3CrudPolicy
```
   
```
@aws
runtime nodejs12.x
policies arn....
```

All of these produce a Policy array inside sam.json with the raw string when `arc package` is run, but when the deployed function is inspected with AWS Lambda console, no extra roles or permissions have been granted.

Needless to say, the code in the function also throws an AccessDenied error when trying to call the AWS SDK s3.listObjectsv2 function.

Our concrete case is needing to grant s3:listObjects on a particular bucket (happens to be in another account, but it's already shared to the Arc account)

arc v9 cli, Ubuntu 20.04 LTS if it helps

**Comment by ryanblock** [comment]
I am able to reproduce this.

Some background: this feature was added a super long time ago (and worked when we added it); it's been ported forward a few times, but since I don't think it's widely used or used by anyone on the core maintainer team, it hasn't been exercised much. So I'm guessing something broke along the way.

In any event, this feature currently follows the resource-based policy implementation as specified here: https://aws.amazon.com/premiumsupport/knowledge-center/lambda-sam-template-permissions/

I'm guessing that since we also use role-based policies for basic least-privilege permissioning, the two policy implementation methods are in conflict.

**Comment by ryanblock** [comment]
Here is the issue: https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-function.html#sam-function-policies

> `Note: If the Role property is set, this property is ignored.`

I am pretty sure this used to not be the case (which is fine) but, as they say, [there's your problem](https://go.discovery.com/tv-shows/mythbusters/videos/theres-your-problem).

**Comment by ryanblock** [comment]
Question: when you specify a policy (example: `policy AmazonDynamoDBFullAccess`), do you expect:
- The normal / default Architect permissions to be applied and overlaid with `AmazonDynamoDBFullAccess`, or
- ONLY `AmazonDynamoDBFullAccess`, with no other default Architect permissions

**Comment by samirrayani** [comment]
I would expect the former:

> The normal / default Architect permissions to be applied and overlaid with AmazonDynamoDBFullAccess, or

AND maybe there is a clear way to handle both?

**Comment by ryanblock** [comment]
Yeah, I think the way to handle both would be for defined policies to be explicit, and you have to opt into the default Architect permissions, which we'd give you as an internal shortcut with something like `_arc` (e.g. `policy _arc AmazonDynamoDBFullAccess`. wdyt?

**Comment by samirrayani** [comment]
oh I dig that! 👍

**Comment by tomchiverton** [comment]
I think when users write "Policy arn::...." they are not expecting it to break unless their policy defines all sorts of things that Arc needs. 
That would be allowing users to shoot themselves in the foot all to easily. What happens when a new Arc version changes the required extra permissions ?

That detail should be managed by Arc.

So the permissions in the Policy should be added to those of whatever internal Role Arc is generating for it's own use.

**Comment by ryanblock** [comment]
We discussed more. Since this is ultimately security-related and can have some extremely dangerous ramifications if we get it wrong, we are going to assume a strong (and explicit) security posture. As such, we'll provide an easy affordance for folks to opt into the default permissions, e.g. `policy _arc AmazonDynamoDBFullAccess`. Stand by for the fix!

**Comment by ryanblock** [comment]
Patches for fix here:
https://github.com/architect/inventory/pull/26
https://github.com/architect/package/pull/140

Fix is available for testing in Arc 9.0.1 RC (`npm i @architect/architect@RC`); the opt-in for the default Architect policies is `architect-default-policies`.

**Comment by ryanblock** [comment]
Fixed in Arc 9.0.1!
